client/src/pages/Reports.js

```diff
@@ -47,6 +50,73 @@ const Reports = () => {
         admin: "container-reportsTable"
     });
 
+    const createAndDownloadPDF = (ReportId) => {
+        setPdfLoading(true);
+        setCurrentPdfLoading(ReportId);
+
+        let header = {
+            'authorization': "Bearer " + cookies.get("accessToken"),
+        }
+
+        let param = {
+            reportid: ReportId
+        }
+
+        Axios.defaults.withCredentials = true;
+
+        Axios.post(`${process.env.REACT_APP_API}/reports/createPdf`, param, { headers: header })
+            .then((response) => {
+                if(response.data === true) {
+                    Axios.get(`${process.env.REACT_APP_API}/reports/fetchPdf`, { params: param, headers: header, responseType: 'arraybuffer' })
+                    .then((res) => {
+                        const pdfBlob = new Blob([res.data], { type: 'application/pdf' });
+
+                        const url = URL.createObjectURL(pdfBlob);
+
+                        var element = document.createElement('a');
+                        document.body.appendChild(element);
+                        element.style = "display: none";
+                        element.href = url;
+                        element.download = `ChartReport-${ReportId}`;
+                        element.click();
+                        document.body.removeChild(element);
+                        setPdfLoading(false);
+                    })
+                    .catch((error) => {
+                        setPdfLoading(false);
+                        if (error.response) {
+                            if (error.response.status === 403 || error.response.status === 401) {
+                                alert("You are not authorized to perform this action.");
+                            }
+                            else {
+                                alert("Could not fetch pdf file...");
+                            }
+                        }
+                        else if (error.request) {
+                            alert("Could not fetch pdf file...");
+                        }
+                    });
+                }
+                else {
+                    alert("Could not reach b&C Engine...");
+                }
+            })
+            .catch((error) => {
+                setPdfLoading(false);
+                if (error.response) {
+                    if (error.response.status === 403 || error.response.status === 401) {
+                        alert("You are not authorized to perform this action.");
+                    }
+                    else {
+                        alert("Could not generate pdf file...");
+                    }
+                }
+                else if (error.request) {
+                    alert("Could not generate pdf file...");
+                }
```

Comment:
Same comment as above

Reply:
What would you recommend I put instead?

Reply:
here when there is an error with the request (`error.request`), we should let the user know by makiing an alert that says 'Could not reach the b&C Engine'